### PR TITLE
Enforces D105 in ExPASy, motifs and SeqIO modules.

### PR DESCRIPTION
**Pull request by Caiofcas**
Also adds missing docstrings to methods in Bio.Align.PairwiseAlignment Class.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

Some progress towards closing #2116 . 

Enforces D105  adding missing magic method docstrings in modules:

- `Bio.SeqIO`
- `Bio.motifs`
- `Bio.ExPASy`
- `Bio.Align`
  - Only for the `PairwiseAlignment` Class.

### CODE REVIEW

File: Bio/Align/__init__.py [hunk 1]
```diff
@@ -947,21 +947,27 @@ def __init__(self, target, query, path, score):
...
     def __eq__(self, other):
+        """Check if two PairwiseAlignment have the same path."""
         return self.path == other.path
 
     def __ne__(self, other):
+        """Check if two PairwiseAlignment have different paths."""
         return self.path != other.path
 
     def __lt__(self, other):
+        """Check if _other_ has a smaller path."""
```
Comment (mdehoon) [inline Bio/Align/__init__.py:958]: Here and in the ones below, It's not so much about a path being smaller / bigger, it's more about the sort order (e.g should self come before other).

File: Bio/Align/__init__.py [hunk 2]
```diff
@@ -2055,6 +2090,7 @@ def __init__(self, seqA, seqB, score, paths):
         self.index = -1
 
     def __len__(self):
+        """Return the number of alignments in self.paths ."""
```
Comment (mdehoon) [inline Bio/Align/__init__.py:2093]: """Return the number of alignments."""

File: Bio/SeqIO/TwoBitIO.py [hunk 3]
```diff
@@ -104,6 +104,7 @@ def __init__(self, stream, offset, length):
         super().__init__()
 
     def __getitem__(self, key):
+        """Return the sequence (as a bytes object) for the requested region."""
```
Comment (mdehoon) [inline Bio/SeqIO/TwoBitIO.py:107]: """Return the sequence contents (as a bytes object) for the requested region."""

File: Bio/SeqIO/TwoBitIO.py [hunk 4]
```diff
@@ -247,4 +250,5 @@ def keys(self):
         return self.sequences.keys()
 
     def __len__(self):
+        """Return number of sequences in TwoBitIterator object."""
```
Comment (mdehoon) [inline Bio/SeqIO/TwoBitIO.py:253]: """Return the number of sequences."""

File: Bio/Align/__init__.py [hunk 5]
```diff
@@ -947,21 +947,27 @@ def __init__(self, target, query, path, score):
...
     def __eq__(self, other):
+        """Check if two PairwiseAlignment objects have the same path."""
         return self.path == other.path
 
     def __ne__(self, other):
+        """Check if two PairwiseAlignment objects have different paths."""
         return self.path != other.path
 
     def __lt__(self, other):
+        """Check if self.path should come before other.path ."""
```
Comment (mdehoon) [inline Bio/Align/__init__.py:958]: """Check if self should come before other."""

**Review by mdehoon — APPROVED** [review]

**Comment by peterjc** [comment]
Thank you - do you want to have your name added too?

**Comment by Caiofcas** [comment]
You mean to `CONTRIB.rst`? I'm already in it because of another PR.

**Comment by peterjc** [comment]
Good - that and the next release's credits.

**Comment by Caiofcas** [comment]
I think I'll add that in #3544 then, I feel it's a more substantial contribution.

**Pull request merged**